#168781633 - Manager Approve Reject Requests

nignanthomas opened the pull request:
#### What does this PR do?
Manager Approve Reject Requests.

#### Description of Task to be completed?
- Create processReqquestAction.
- Create ProcessRequest reusable button component.
- Create tests.

#### How should this be manually tested?
- Clone ft-approve-reject-requests-168781633 branch
- run `npm i`
- run `npm run dev`
- Login as a Manager and click on `Manage Requests` in the Menu

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168781633](https://www.pivotaltracker.com/story/show/168781633)

#### Screenshots (if appropriate)
<img width="1084" alt="Screen Shot 2019-12-31 at 17 25 58" src="https://user-images.githubusercontent.com/42959540/71625956-469d9080-2bf3-11ea-8bd0-a6c6ed46fb5f.png">
<img width="1085" alt="Screen Shot 2019-12-31 at 17 45 35" src="https://user-images.githubusercontent.com/42959540/71626358-6635b880-2bf5-11ea-9393-9b546d3ea23c.png">

#### Questions:
N/A

nignanthomas committed:
feature(approve-reject-request): Manager Approve Reject Requests

- create processReqquestAction

- create ProcessRequest reusable button component

- create tests

[Finishes #168781633]

ngireric123 commented on src/components/pages/requests/ProcessRequest.js:
```diff
@@ -0,0 +1,48 @@
+import { processRequestAction, singleRequestAction } from '../../../actions/requestsActions';
+import { getManagerRequestAction } from '../../../actions/managerRequestAction';
+
+
```
remove one line

ngireric123 commented on src/styles/requests/requests.scss:
```diff
@@ -31,4 +31,7 @@
+}
+.process-request-button {
+  text-transform: uppercase;
 }
```
add new line

ngireric123 commented on src/tests/reducersTests/processRequestReducerTest.spec.js:
```diff
@@ -0,0 +1,50 @@
+    });
+  });
+
+});
```
add new line

ngireric123 reviewed (approved):
LGTM

sabin18 reviewed (changes requested):
Good job @nignanthomas , put approve and reject button below the view request button to make it look good.

nakiwuge commented on src/components/pages/requests/ProcessRequest.js:
```diff
@@ -0,0 +1,47 @@
+
+  processAction = async () => {
+    const { props } = this;
+    const { action, id } = props;
```
Add a new line

nakiwuge commented on src/components/pages/requests/ProcessRequest.js:
```diff
@@ -0,0 +1,47 @@
+
+  render() {
+    const { isLoading } = this.state;
+    const { action, variant } = this.props;
```
Add a new line

nakiwuge reviewed (changes requested):
Good job @nignanthomas. Kindly work on the feedback below.
- The user should receive a confirmation message when the click `Approve/Reject`. Something like `Are you sure you want to approve/reject this request?`. This should be a reusable component because it will also be used on DELETE.
- Disable the `Approve` button when the loader is on the `Reject` button and vice versa
- Go through other comments I have left on the pr

nignanthomas committed:
feature(approve-reject-request): Manager Approve Reject Requests

    - create processReqquestAction

    - create ProcessRequest reusable button component

    - create tests

    [Finishes #168781633]
